sisyphus/simple_linux_utility_for_resource_management_engine.py

```diff
@@ -86,18 +86,23 @@ def system_call(self, command, send_to_stdin=None):
         """
         if self.gateway:
             escaped_command = [shlex.quote(s) for s in command]  # parameters need to be shell safe when sending via ssh
-            system_command = ["ssh", "-x", self.gateway, "-o", "ConnectTimeout", gs.WAIT_PERIOD_BETWEEN_CHECKS] + [
+            system_command = ["ssh", "-x", self.gateway, "-o", "BatchMode=yes"] + [
                 " ".join(["cd", os.getcwd(), "&&"] + escaped_command)
             ]
         else:
             # no gateway given, skip ssh local
             system_command = command
 
         logging.debug("shell_cmd: %s" % " ".join(system_command))
-        p = subprocess.Popen(system_command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         if send_to_stdin:
             send_to_stdin = send_to_stdin.encode()
-        out, err = p.communicate(input=send_to_stdin, timeout=gs.WAIT_PERIOD_BETWEEN_CHECKS)
+        p = subprocess.run(
+            system_command,
+            stdin=send_to_stdin,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            timeout=gs.WAIT_PERIOD_BETWEEN_CHECKS,
+        )
```

Comment:
```suggestion
        p = subprocess.run(
            system_command,
            input=send_to_stdin,
            capture_output=True,
            timeout=gs.WAIT_PERIOD_BETWEEN_CHECKS,
        )
```

Reply:
Why were stdout/err a pipe to begin with? I thought that was needed by sisyphus :thinking:

Reply:
Because you need to check stdout/err afterwards? This must be there. But better use `capture_output=True` (shorter/simpler). But also, using `stdin` here is wrong. You need to use `input` instead.

Reply:
Okay, that makes sense. I think I've addressed all changes, resolving thread.